FitGoal/FitGoal/UserProfileConfiguration/ViewControllers/UserInfoEntryViewDelegate.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  UserInfoEntryViewDelegate.swift
+//  FitGoal
+//
+//  Created by Eliany Barrera on 25/5/20.
+//  Copyright © 2020 Eliany Barrera. All rights reserved.
+//
+
+import UIKit
+
+class UserInfoEntryViewDelegate: NSObject, IconListViewDelegate,  UITextFieldDelegate {
+    
+    private let appPreferences = AppPreferences()
+    private let userProfileType: UserProfileType
+    
+    init(userInfoEntryView: UIView, userProfileType: UserProfileType) {
+        self.userProfileType = userProfileType
+        super.init()
+        setDelegate(forView: userInfoEntryView)
+    }
+    
+    private func setDelegate(forView userInfoEntryView: UIView) {
+        if let userInfoEntryView = userInfoEntryView as? UITextField {
+            userInfoEntryView.delegate = self
+        } else if let userInfoEntryView = userInfoEntryView as? IconListView {
+            userInfoEntryView.delegate = self
+        }
+    }
+    
+    func iconListView(_ listView: IconListView, didSelectIcon iconView: WalkthroughIconView) {
+        switch userProfileType {
+        case .fitnessGoal:
+            appPreferences.fitnessGoal = iconView.icon.name
+        case .fitnessLevel:
+            appPreferences.fitnessLevel = iconView.icon.name
+        case .gender:
+            appPreferences.userGender = iconView.icon.name
+        default:
+            assertionFailure("ListIconViewDelegate shouldn't be called for views that are not listIconView")
```

Comment:
The message of this assertion is wrong. If you get to this point the reason isn't the the view isn't a `ListIconView`.

Reply:
true